src/packagedcode/opam.py

```diff
@@ -0,0 +1,354 @@
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+import io
+import logging
+import re
+
+import attr
+from packageurl import PackageURL
+
+from commoncode import filetype
+from commoncode import fileutils
+from packagedcode import models
+
+
+"""
+Handle opam package.
+"""
+
+TRACE = False
+
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+@attr.s()
+class OpamPackage(models.Package):
+    metafiles = ('*opam',)
+    extensions = ('.opam',)
+    default_type = 'opam'
+    default_primary_language = 'Ocaml'
+    default_web_baseurl = 'https://opam.ocaml.org/packages'
+    default_download_baseurl = None
+    default_api_baseurl = 'https://github.com/ocaml/opam-repository/blob/master/packages'
+
+    @classmethod
+    def recognize(cls, location):
+        yield parse(location)
+
+    @classmethod
+    def get_package_root(cls, manifest_resource, codebase):
+        return manifest_resource.parent(codebase)
+
+    def repository_homepage_url(self, baseurl=default_web_baseurl):
+        if self.name:
+            return '{}/{}'.format(baseurl, self.name)
+
+    def api_data_url(self, baseurl=default_api_baseurl):
+        if self.name and self.version:
+            return '{}/{}/{}.{}/opam'.format(baseurl, self.name, self.name, self.version)
+
+
+def is_opam(location):
+    return location.endswith('opam')
+
+
+def parse(location):
+    """
+    Return a Package object from a opam file or None.
+    """
+    if not is_opam(location):
+        return
+
+    opams = parse_opam(location)
+    return build_opam_package(opams)
+
+
+def build_opam_package(opams):
+    """
+    Return a Package from a opam file or None.
+    """
+    package_dependencies = []
+    deps = opams.get('depends') or []
+    for dep in deps:
+        package_dependencies.append(
+            models.DependentPackage(
+                purl=dep.purl,
+                requirement=dep.version,
+                scope='dependency',
+                is_runtime=True,
+                is_optional=False,
+                is_resolved=False,
+            )
+        )
+
+    name = opams.get('name')
+    version = opams.get('version')
+    homepage_url = opams.get('homepage')
+    download_url = opams.get('src')
+    vcs_url = opams.get('dev-repo')
+    bug_tracking_url = opams.get('bug-reports')
+    declared_license = opams.get('license')
+    sha1 = opams.get('sha1')
+    md5 = opams.get('md5')
+    sha256 = opams.get('sha256')
+    sha512 = opams.get('sha512')
+    summary = opams.get('synopsis')
+    description = opams.get('description')
+    if summary:
+        description = summary
+    elif summary and description:
+        if len(summary) > len(description):
+            description = summary
```

Comment:
What about instead always using both?
the convention used elsewhere is to combine summary + description in a single text (and avoiding redundancies too).
See https://github.com/nexB/scancode-toolkit/blob/62fc91d64d59aac202ebe610c7e2ce46ae70b23f/src/packagedcode/rubygems.py#L294
This could be refined to not include the summary/short desc part the long description already starts with the shorter summary.

Reply:
Yes, this will be good idea. :)